### PR TITLE
Add Spotify client

### PR DESCRIPTION
**Pull request by skasturi**
Added Spotify client

### CODE REVIEW

**Comment by skasturi** [comment]
Added Yahoo, Spotify.

File: OAuth2/Client/Impl/YahooClient.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+namespace OAuth2.Client.Impl
+{
+    /// <summary>
+    /// Yahoo client 
+    /// Right now only Yahoo Gemini and Yahoo Social support OAuth2
+    /// https://developer.yahoo.com/oauth2/guide/
+    /// </summary>
+    public class YahooClient : OAuth2Client
+    {
+        public string userProfileGUID;
```
Comment (titarenko) [inline OAuth2/Client/Impl/YahooClient.cs]: Pull request looks great, but could you please leave this field private and provide access to its value via property with getter?

**Comment by titarenko** [comment]
I've sent you a small PR on this PR (sounds weird, but is it as is :) with a little change. Please accept it (if it's OK) and we will finish with merge.

**Comment by skasturi** [comment]
Done.

**Comment by titarenko** [comment]
Great! Thank you for contribution!

**Pull request merged**